DASH-IF-IOP.bs.md

```diff
@@ -10,7 +10,242 @@ Live discussion in #document-authoring on Slack.
+
+Note: Inaccurate addressing is intended to allow reasoning on the [=MPD timeline=] using average values for [=media segment=] timing. If the addressing data says that a [=media segment=] contains 4 seconds of data on average, a client can predict with reasonable accuracy which samples are found in which segments, while at the same time the packager is not required to emit per-segment timing data in the [=MPD=]. It is expected that the content is packaged with this contraint in mind (i.e. **every** segment cannot be inaccurate in the same direction - a shorter segment now implies a longer segment in the future to make up for it).
+
+<div class="example">
+Consider a [=media segment=] with a nominal start time of 10 seconds from period start and a nominal duration of 4 seconds, within a [=period=] of unlimited duration, using an addressing mode that allows for inaccuracy.
+
+The following are all valid contents for such a media segment:
+
+* samples from 10 to 14 seconds (perfect accuracy)
+* samples from 8 to 16 seconds (maximally large segment allowed by drift tolerance, 50% increase from both ends)
+* samples from 11.9 to 12 seconds (near-minimally small segment; while drift tolerance allows 50% decrease from both ends, resulting in zero duration, every segment must still contain at least one sample)
+* samples from 8 to 12 seconds (maximal drift toward zero point at both ends)
+* samples from 12 to 16 seconds (maximal drift away from zero point at both ends)
+
+Near [=period=] boundaries, all the constraints of timing and addressing must still be respected. Consider a [=media segment=] with a nominal start time of 0 seconds from [=period=] start and a nominal duration of 4 seconds, using an addressing mode that allows for inaccuracy.
+
+If such a [=media segment=] contained samples from 1 to 5 seconds (drift of 1 second away from zero point at both ends, which is within acceptable limits) it would be non-conforming because it would leave a gap of 1 second at the start of the [=period=] that is not covered by samples.
+</div>
+
+## Segment addressing modes ## {#timing-addressingmodes}
+
+Issue: Determine appropriate content for this section.
+
+## Segment alignment ## {#timing-segmentalignment}
+
+Segments are said to be aligned if the start/end points of all [=media segments=] are equal in all [=representations=] that belong to the same adaptation set. This is expressed by the `AdaptationSet@segmentAlignment` or `AdaptationSet@subsegmentAlignment` attributes in the [=MPD=] (depending on profile).
+
+Segments SHALL be aligned.
+
+## Period continuity ## {#timing-continuity}
```

Comment:
> The case where the periods are immediately next to one another probably doesn't happen much in practice, because it's unlikely a new period would need to be started mid-segment in that case.

That nonconforming->conforming aspect is one place where this occurs but it also has very real scenarios in other situations. Some examples of when periods with continuous representations will be right next to each other:

* the set of audio/text languages changes (as may be with live TV, whenever a different title starts) - the video representations and any non-modified representations will remain continuous
* encryption/DRM parameters change due to scheduled key rotation or re-authorization trigger

> So it seems to me like this section is describing how a client that isn't able to implement clipping at period boundaries
Right. If a player can clip at a sample level (which might mean even starting the next period at  a non-keyframe!) then it can do it with perfect accuracy and has no need to worry about deduplication.

> should special-case a scenario that's unlikely to occur,
It is not clear to me what you refer to as the unlikely scenario.

Do you consider imperfectly clipping clients the unlikely case? I lack a good overview of client capabilities but it seems prudent not to assume that sample-based clipping is always available, so I did not build such an assumption into the PR.

Or do you consider splitting a period in the middle of segments the unlikely case? It seems to me that splitting a period in the middle of a segment is the more likely case as scheduling decisions tend to be clock-driven, not segment-driven, not to mention that different adaptation sets may have different segmentation with potentially no overlapping segment boundaries even if you wanted to cut at a boundary.

> to avoid playing out the segment twice. And even then saying that what ends up happening is going to be implementation specific
Agreed. It is to avoid playing out the segment twice if the player cannot do sample-accurate clipping and it is going to be implementation specific.

Reply:
> It is not clear to me what you refer to as the unlikely scenario.

I'm referring to the scenario in which it's necessary to start a new period mid-segment. Given the other examples you've described, I'll take that back :). So feel free to disregard this comment!

> Do you consider imperfectly clipping clients the unlikely case?

In general across all clients, I don't know.

I can say something about ExoPlayer though: The way ExoPlayer is implemented currently, each period is treated pretty independently and there's a fairly hard boundary at quite a high level between the objects corresponding to one period and those corresponding to the next. So it would require a lot of refactoring and probably changing some fairly fundamental assumptions for us to handle the case described in a special way. ExoPlayer does implement clipping. I'm not sure we clip perfectly (we might drop or repeat a single sample at the boundary), but we probably get pretty close. I think the conclusion would be that it wont be worth us doing anything special to handle the case described in this section. In other words, we'll rely on the clipping being close enough. If it's not close enough, it would probably be easier for us to fix the clipping than do something clever around detecting that a period starts with a segment that the previous period ended with.